mule-extensions-soap-api/src/main/java/org/mule/runtime/extension/api/soap/SoapOutputPayload.java

```diff
@@ -67,17 +69,24 @@ public String toString() {
     try {
       String hs = headers.values().stream().map(v -> "\"" + v.getValue() + "\"").collect(joining(",\n  "));
       String as = String.join(", ", attachments.keySet());
-      String bodyString = IOUtils.toString(body.getValue(), getBodyCharset());
       return "{\n" +
-          "body:" + bodyString + ",\n" +
+          "body:" + getBodyString() + ",\n" +
           "headers: [" + hs + "]" + ",\n" +
           "attachments: [" + as + "]" + "\n" +
           "}";
-    } catch (IOException e) {
+    } catch (Exception e) {
+      LOGGER.error("Error building SoapResponse string: " + e.getMessage(), e);
       return "Error building SoapResponse string";
     }
   }
 
+  private String getBodyString() throws IOException {
+    Object value = body.getValue();
+    InputStream inputStream =
+        value instanceof CursorStreamProvider ? ((CursorStreamProvider) value).openCursor() : ((InputStream) value);
```

Comment:
you can't trust to be an InputStream, do also the `instanceof` is not do a simple `toString()`